src/components/ImmutableDevices/ImmutableDevices.cy.js

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import { Route, Routes } from 'react-router-dom';
+import ImmutableDevices from './ImmutableDevices';
+import {
+  featureFlagsInterceptors,
+  hostsInterceptors,
+} from '../../../cypress/support/interceptors';
+
+const defaultProps = {
+  mergeAppColumns: (columns) => columns,
+};
+
+// eslint-disable-next-line react/prop-types
+const MockRouter = ({ path = '/insights/inventory', ...props }) => (
+  <Routes>
+    <Route path={path} element={<ImmutableDevices {...props} />} />
+    <Route path={'*'} element={<div id="mock-detail-page" />} />
```

Comment:
Do we really need the second route? I see it is used in one of the tests to check that browser navigated to a new URL. But can we just assert on `window.location.pathname`? This would help us remove extra code and just stick to `mountWithContext` function.

Reply:
Wondering if having this extra code deserves to be kept. Because, instead of relying on bare window location, I will be sure that the next tab is open. What do you think?